lib/common/tests/strings/pcmk__char_in_any_str.c

```diff
@@ -0,0 +1,44 @@
+#include <glib.h>
+
+#include <crm_internal.h>
+
+static void
+empty_list(void)
+{
+    g_assert_false(pcmk__char_in_any_str('x', NULL));
+    g_assert_false(pcmk__char_in_any_str('\0', NULL));
```

Comment:
Understandable concern but I think `g_assert_(true|false)` is the way to go here.

1. g_assert() and g_assert_true() run the same check (G_LIKELY(expr)) if G_DISABLE_ASSERT is not set. The only differences are the behavior if G_DISABLE_ASSERT is set, and a slight log message difference.

```
#define g_assert_true(expr)             G_STMT_START { \
                                             if G_LIKELY (expr) ; else \
                                               g_assertion_message (G_LOG_DOMAIN, __FILE__, __LINE__, G_STRFUNC, \
                                                                    "'" #expr "' should be TRUE"); \
                                        } G_STMT_END
#define g_assert_false(expr)            G_STMT_START { \
                                             if G_LIKELY (!(expr)) ; else \
                                               g_assertion_message (G_LOG_DOMAIN, __FILE__, __LINE__, G_STRFUNC, \
                                                                    "'" #expr "' should be FALSE"); \
                                        } G_STMT_END
...
#else /* !G_DISABLE_ASSERT */
...
#define g_assert(expr)                  G_STMT_START { \
                                             if G_LIKELY (expr) ; else \
                                               g_assertion_message_expr (G_LOG_DOMAIN, __FILE__, __LINE__, G_STRFUNC, \
                                                                         #expr); \
                                        } G_STMT_END
```

2. The big-endian issue that we hit with glib option parsing was due to some evil-ness in the way it assigns to arguments. It takes a pointer to the store variable and casts it to a pointer to gboolean when assigning. Since a gboolean is 4 bytes and a bool is 1 byte, we lose the least significant 3 bytes of the rvalue during assignment, when arg_data is a bool variable.
```
          *(gboolean *)entry->arg_data = !(entry->flags & G_OPTION_FLAG_REVERSE);
```

A normal cast from gboolean to bool (or vice-versa) without pointer casting should be safe. A gboolean TRUE is non-zero, so it will be cast to `true` for a bool variable. A gboolean FALSE is zero, so it will be cast to `false`.

Reply:
I wasn't thinking of g_assert() but g_assert_cmpint(..., ==, false). I understand that mixing is not harmful in this case, but I'm leaning toward enforcing consistency to keep it on developer's minds when they use one or the other.

There was a time when Pacemaker used gboolean almost exclusively, but bool crept in, and I decided to prefer bool except when used with glib functions, mainly because bool is standard, and I prefer not to force public-API-using external code to include glib.h if there's an easy way around it. But I'm second-guessing that decision. I lean toward keeping it, but not mixing the models.

Reply:
Hmm, yes I see I misread. I'm happy to re-push to use g_assert_cmpint(). I'm struggling to think of when it would ever make a practical difference one way or the other, except for consistency. So I have no strong feelings; I defaulted to using what fit semantically.

Reply:
Come to think of it, another reason I went with g_assert_true/false is your comment in #2137 ;)

> I find ! more readable than ==false for boolean comparisons -- otherwise my mind has to backtrack. Also it's consistent with not using ==true.

Reply:
Ah yes but that doesn't apply because I'll never read the tests again ;)

I'm fine with it either way. The consistency is indeed the only benefit, but consistency can sometimes be a help in a 150K line code base.